lib/ansible/modules/cloud/amazon/GUIDELINES.md

```diff
@@ -240,36 +245,45 @@ Boto3 provides lots of useful information when an exception is thrown so pass th
 along with the message.
 
 ```python
-# Import ClientError from botocore
+from ansible.module_utils.ec2 import HAS_BOTO3
+
 try:
-    from botocore.exceptions import ClientError
-    HAS_BOTO3 = True
+    import botocore
 except ImportError:
-    HAS_BOTO3 = False
+    pass  # caught by imported HAS_BOTO3
 
 # Connect to AWS
 ...
 
 # Make a call to AWS
+name = module.params.get['name']
 try:
-    result = connection.aws_call()
-except ClientError as e:
-    module.fail_json(msg=e.message, exception=traceback.format_exc(),
+    result = connection.describe_frooble(FroobleName=name)
+except botocore.exceptions.ClientError as e:
+    module.fail_json(msg="Couldn't obtain frooble %s: %s" % (name, str(e)),
+                     exception=traceback.format_exc(),
                      **camel_dict_to_snake_dict(e.response))
 ```
 
+Note: we use `str(e)` rather than `e.message` as the latter doesn't
```

Comment:
Should we really be using to_native()?

Reply:
I don't know - in what circumstances would `str(e)` cause a problem? (`str(e)` appears to work for both python2 and 3)

Reply:
This question also applies my second commit that I've just pushed to handle `ProfileNotFound`